src/groups/special_galilean_group.jl

```diff
@@ -0,0 +1,198 @@
+Given group element `g = (R, v)` and event `h = (x, t)`, computes the transformed event `k = (R*x + v*t, t)`.
+See [Kelly:2025; section 4.1](@cite).
+"""
+function LieGroups.apply!(A::GroupAction{RotationBoostAction}, k, g, h)
+    G = A.group
+    R = submanifold_component(G, g, Val(1))
+    v = submanifold_component(G, g, Val(2))
+
+    H = A.manifold
+    p = submanifold_component(H, h, Val(1))
+    t = submanifold_component(H, h, Val(2))[1]
+
+    k_p = submanifold_component(H, k, Val(1))
+    k_p .= v * t + R * p
+
+    k_t = submanifold_component(H, k, Val(2))
+    k_t .= t
+    return k
+end
+
+const LeftSpecialGalileanGroupOperation = LeftSemidirectProductGroupOperation{
+    LeftSemidirectProductGroupOperation{
+        MatrixMultiplicationGroupOperation, AdditionGroupOperation, LeftMultiplicationGroupAction, ActionActsOnRight,
+    },
+    ProductGroupOperation{Tuple{AdditionGroupOperation, AdditionGroupOperation}},
+    RotationBoostAction,
+    ActionActsOnRight,
+}
+
+const SpecialGalileanGroup{T} = LieGroup{
+    ℝ,
+    <:LeftSpecialGalileanGroupOperation,
+    <:ProductManifold{
+        ℝ,
+        Tuple{
+            <:ProductManifold{ℝ, Tuple{<:Rotations{T}, <:Euclidean{T, ℝ}}},
+            <:ProductManifold{
+                ℝ,
+                Tuple{
+                    <:Euclidean{T, ℝ}, <:Euclidean{ManifoldsBase.TypeParameter{Tuple{1}}, ℝ},
+                },
+            },
+        },
+    },
+}
+
+"""
+    SpecialGalileanGroup(n::Int)
+
+Construct the special Galilean group SGal(n) as a nested semidirect product:
+    ``(SO(n) ⋉ ℝⁿ) ⋉ (ℝⁿ × ℝ)``
+where ``R ∈ SO(n)`` are spatial rotations, ``v ∈ ℝⁿ`` are velocity boosts, and ``(p, t) ∈ (ℝⁿ × ℝ)`` are the (position, time) events.
+The affine representation of the group is given by the matrix:
+```math
+\\mathrm{SGal}(3) = \\begin{bmatrix}
+R & v & p \\\\
+0 & 1 & t \\\\
+0 & 0 & 1
+\\end{bmatrix} \\subset \\mathbb{R}^{5\\times 5}
+```
+And the ArrayPartition representation as:
+``((R, v), (p, t))``
+
+!!! note "Technical Detail"
+    The ArrayPartition (default) implementation requires `RecursiveArrayTools.jl` to be loaded.
+
+[Kelly:2025](@cite)
+"""
+function SpecialGalileanGroup(n::Int)
+    G = SpecialOrthogonalGroup(n) ⋉ TranslationGroup(n)
+    N = TranslationGroup(n) × TranslationGroup(1)
+    return LieGroup(
+        ProductManifold(G.manifold, N.manifold),
+        LeftSemidirectProductGroupOperation(G.op, N.op, RotationBoostAction(), ActionActsOnRight())
+    )
+end
+
+#
+#
+# doc strings
+
+_doc_SGal3_exp = """
+    LieGroups.exp(M::SpecialGalileanGroup, X)
+    LieGroups.exp!(M::SpecialGalileanGroup, h, X)
+
+Compute the Lie group exponential function on the [`SpecialGalileanGroup`](@ref)`(3)`,
+where `X` is an element of the Lie algebra represented as an `ArrayPartition`.
+
+The closed-form expression for the matrix exponential from [Kelly:2025; section 6](@cite) is used.
```

Comment:
How complicated is the formula? Would it maybe make sense to write them there for completeness? For not-too-complicated formulae we do that throughout our packages.

Reply:
We basically need these:
<img width="826" height="322" alt="image" src="https://github.com/user-attachments/assets/a00dbc12-6a3e-4bf3-baa0-9fdb4a2d484a" />

Are they too complicated or should I also add them?

Reply:
If you do bot mind, please add them :)